src/main.rs

```diff
@@ -13,29 +13,27 @@ fn main() {
             let mut buf = String::new();
             io::stdin().read_line(&mut buf).unwrap();
             let input = buf.trim();
+
             if input == "exit" {
                 process::exit(0);
             }
             if input == "restart" {
                 break;
             }
-
             if input.len() != 4 {
                 println!("Please enter a 4-characters word");
                 continue;
             }
 
-            if service.verify(input) {
-                println!("Congratulations, you have guessed the word!");
-                break;
-            } else {
-                println!("{}\n", service.hint());
+            match service.verify(input) {
+                GuessResult::Guessed => {
+                    println!("Congratulations, you have guessed the word!");
+                    break;
+                }
+                GuessResult::Missed { hint } => {
+                    println!("{}\n", hint);
```

Comment:
You can do 
```suggestion
                    println!("{hint}\n");
```

Back in the days, it was not possible, so a lot of tutorials still don't show this properly.